lib/improver/spotdata/site_data.py

```diff
@@ -0,0 +1,246 @@
+            utckey = 'utcoffset'
+        elif 'utc_offset' in site_data[0].keys():
+            utckey = 'utc_offset'
+
+        # Use available UTC offset data, or set data with crude longitude
+        # derived value if it is missing.
+        if utckey is None:
+            self.utc_offsets = set_utc_offset(self.longitudes)
+        else:
+            self.utc_offsets = np.array([site[utckey] for site in site_data])
+
+        return self.construct_site_dictionary()
+
+    def runtime_list(self, latitudes, longitudes,
+                     altitudes=None, site_ids=None):
+        """
+        Use data provided on the command line/controlling suite at runtime.
+
+        Args:
+        -----
+        latitudes/longitudes : lists of floats
+            Lists of coordinates at which to extract data, these define the
+            positions of the SpotData sites.
+
+        altitudes : list of floats
+            List of altitudes that can optionally be used in defining sites
+            on the fly. If unset the altitudes will be assumed as equivalent
+            to which ever neighbouring grid point is used as a data source.
+
+        site_ids : list of integers
+            Optional integer site_ids that will be used as keys for the site
+            information in the returned site data dictionary. If unset a simple
+            counting index is used.
+
+        Returns:
+        --------
+        sites : dict
+            Dictionary of site data.
+
+        """
+        if len(latitudes) != len(longitudes):
+            raise ValueError('Unequal number of latitude and longitudes '
+                             'given for sites.')
+        if ((altitudes is not None and len(altitudes) != len(latitudes)) or
+                (site_ids is not None and len(site_ids) != len(latitudes))):
+            raise ValueError('Unequal number of altitudes/site_ids and '
+                             'latitude/longitudes given for sites.')
+
+        if site_ids is not None:
+            self.site_ids = np.array(site_ids)
+        else:
+            self.site_ids = np.arange(len(latitudes))
+        if altitudes is not None:
+            self.altitudes = np.array(altitudes)
+        else:
+            self.altitudes = np.full(len(latitudes), -9999)
+
+        self.latitudes = np.array(latitudes)
+        self.longitudes = np.array(longitudes)
+        self.utc_offsets = set_utc_offset(self.longitudes)
+
+        return self.construct_site_dictionary()
+
+    def construct_site_dictionary(self):
+        """
+        Constructs a dictionary of site data regardles of source to give the
+        spotdata routines a consistent source of site data.
+
+        Returns:
+        --------
+        sites : Dictionary of site data.
+
+        """
+        sites = OrderedDict()
+        for i_site, site_id in enumerate(self.site_ids):
+            if self.utc_offsets[i_site] is None:
+                self.utc_offsets[i_site] = 0
```

Comment:
It will never reach this as far as I can see, but I didn't want anything unset to float through. I'll delete this statement as it should not be needed.